tool.go

```diff
@@ -91,6 +91,10 @@ func main() {
 	cmdBuild.Flags().AddFlag(flagTags)
 	cmdBuild.Flags().AddFlag(flagLocalMap)
 	cmdBuild.Run = func(cmd *cobra.Command, args []string) {
+		if err := verifyAndUnsetGOARCH(); err != nil {
```

Comment:
cmd/go does this as part of cmd/go, rather than compiler. But only for `gc`, it seems.

https://github.com/golang/go/blob/master/src/cmd/go/internal/work/build.go#L1098-L1101

I'll try and see it if works out.